From 5260e832da20433f22f7fb926a279ca3e94a95a7 Mon Sep 17 00:00:00 2001
From: cjihrig <cjihrig@gmail.com>
Date: Sun, 14 Oct 2018 11:55:16 -0400
Subject: [PATCH] unix: restore skipping of phys_addr copy

Prior to https://github.com/libuv/libuv/pull/1999,
uv_interface_addresses() did not populate the phys_addr field on
Cygwin or MinGW. This commit restores that behavior, which was
mistakenly changed.
---
 src/unix/bsd-ifaddrs.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/unix/bsd-ifaddrs.c b/src/unix/bsd-ifaddrs.c
index 9825b1c4d..2b5c86880 100644
--- a/src/unix/bsd-ifaddrs.c
+++ b/src/unix/bsd-ifaddrs.c
@@ -121,6 +121,7 @@ int uv_interface_addresses(uv_interface_address_t** addresses, int* count) {
     address = *addresses;
     memset(address->phys_addr, 0, sizeof(address->phys_addr));
 
+#if !defined(__CYGWIN__) && !defined(__MSYS__)
     for (i = 0; i < *count; i++) {
       if (strcmp(address->name, ent->ifa_name) == 0) {
         struct sockaddr_dl* sa_addr;
@@ -129,6 +130,7 @@ int uv_interface_addresses(uv_interface_address_t** addresses, int* count) {
       }
       address++;
     }
+#endif
   }
 
   freeifaddrs(addrs);
